sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -76,7 +75,7 @@
     private final PriorityQueue<WeightedDeliveryTag> pendingSendsQueue =
         new PriorityQueue<>(1000, new DeliveryTagComparator());
     private final ClientLogger logger = new ClientLogger(ReactorSender.class);
-    private final ReplayProcessor<AmqpEndpointState> endpointStates;
+    private final Flux<AmqpEndpointState> endpointStates;
```

Comment:
I'm seeing this, `Flux<AmqpEndpointState>`, in many classes, is there a way we could extract this out into a single location where we instantiate it as a Sink?

Reply:
So handlers propagate up an endpoint state, proton-j `EndpointState`. Senders, receivers, sessions, and connections can all have different states... and based on each state, we want to either close child links, dispose deliveries, etc. So we subscribe to that handler endpoint, do some sort of processing to change it to an azure-core-amqp AmqpEndpointState and then surface that.